supervisor/addons/addon.py

```diff
@@ -166,6 +172,13 @@ def state(self, new_state: AddonState) -> None:
         if new_state == AddonState.STARTED or old_state == AddonState.STARTUP:
             self._startup_event.set()
 
+        # Dismiss boot failed issue if present and we started
+        if (
+            new_state == AddonState.STARTED
+            and self.boot_failed_issue in self.sys_resolution.issues
```

Comment:
I followed the same pattern we had used in `Mounts`. But yea I guess I could cache the object for reuse? The only weird part of that would be the UUID would always be identical but I guess that's not really a big deal, we don't use that for anything other then referencing it in the API anyway.

Reply:
Yeah it does feel weird to me to create an object just to check if it's in there. But since we use that pattern already, let's go with that for now.

Maybe worth a follow up which does this a bit more efficient.